### PR TITLE
"Not Insured" Improper Hangups Fix

### PR DESCRIPTION
**Pull request by MTS11648**
This is a REALLY old issue, but it looks like one that has never been addressed. I remember this being a big lead killer back in the day and (unless this was already fixed in another manner), it probably still is today. This check makes sure that we are on the insurance provider question before hanging up on someone saying "no insurance". (Addresses #30 and #31).

### CODE REVIEW

File: AutoBotCSharp/Agent.cs [hunk 1]
```diff
@@ -4155,14 +4155,16 @@ public string badSpeechParser(string oldSpeech)
...
                 {
-
-                    currentlyRebuttaling = true;
-                    hasAsked = true;
-                    clip = @"C:\SoundBoard\Cheryl\REBUTTALS\sorry.mp3";
-                    Conversation.RollTheClip(clip);
-                    Conversation.RollTheClip(@"C:\SoundBoard\Cheryl\WRAPUP\Have a great day.mp3");
-                    HangUpandDispo("No Insurance");
-                    return false;
+                    if (AutoFormQuestion == Agent.PROVIDER || AutoFormQuestion == Agent.INS_EXP || AutoFormQuestion == Agent.INS_START)
```
Comment (SpadesNeil) [inline AutoBotCSharp/Agent.cs]: Why not just add this in the _else if_ statement above?

Like this:
else if ((AutoFormQuestion == This || AutoFormQuestion == That) && (response.Contains("something") || response == "something else"))

Comment (MTS11648) [inline AutoBotCSharp/Agent.cs]: @SpadesNeil I was going to do that, but it is just easier to read like this IMO.

Comment (SpadesNeil) [inline AutoBotCSharp/Agent.cs]: A little easier to read, but I'm going to be nitpicky about it. It's not a good coding practice. If someone goes to look at this code later they're going to see the "if" statement and think, "Wait, then what's supposed to happen when the if statement isn't met?" or alternatively they try to add or edit conditions that are already covered by the if statement, or they edit the wrong statement.

tl;dr it'll be more confusing down the line

Comment (MTS11648) [inline AutoBotCSharp/Agent.cs]: I suppose I could see that; either way it doesn't really matter. I'll change it to use that format when I get a free chance.
